rs/execution_environment/tests/backtraces.rs

```diff
@@ -0,0 +1,45 @@
+use candid::Encode;
+use ic_registry_subnet_type::SubnetType;
+use ic_state_machine_tests::{StateMachine, StateMachineBuilder};
+use ic_types::{CanisterId, Cycles};
+
+const B: u128 = 1_000 * 1_000 * 1_000;
+
+fn env_with_backtrace_canister() -> (StateMachine, CanisterId) {
+    let wasm = canister_test::Project::cargo_bin_maybe_from_env("backtrace_canister", &[]);
+
+    let env = StateMachineBuilder::new()
+        .with_subnet_type(SubnetType::Application)
+        .build();
+
+    let initial_cycles = Cycles::new(1_000_000 * B);
+    let canister_id = env
+        .install_canister_with_cycles(wasm.bytes(), vec![], None, initial_cycles)
+        .unwrap();
+
+    (env, canister_id)
+}
+
+#[test]
+fn backtrace_test_unreachable() {
+    let (env, canister_id) = env_with_backtrace_canister();
+    assert!(env
+        .execute_ingress(canister_id, "unreachable", Encode!(&()).unwrap())
```

Comment:
These tests all have the same assertion.  

We've had some experiences where underspecified errors in tests end up missing a change in behavior, because the test will error out for a different unrelated reason.

I would recommend looking for a particular error behavior for each of these (different message, code, or whatever is relevant and stable)

Reply:
I should have made it clearer in the description - these tests will be used to assert that the canister backtrace is present in the error message, but that feature is still in progress: https://dfinity.atlassian.net/browse/EXC-1713. For the moment these tests are just asserting that there is an error and as that feature is completed each test will be expanded to check the backtrace. That is the specific error behavior these tests will care about.